action.yaml

```
@@ -0,0 +1,13 @@
+name: Setup go env using go version from go.mod
+description: Fetch go version from go.mod file, set it as an env var and use it to setup go env
+
+runs:
+  using: composite
+  steps:
+    - name: Set GO_VERSION
+      shell: bash
+      run: echo "GO_VERSION=$(grep -m 1 -r 'go' go.mod | awk '{print $2}')" >> $GITHUB_ENV
+    - name: setup go env
```

https://grammarist.com/spelling/set-up-vs-setup/#:~:text=Set%20Up%20%E2%80%93%20What's%20the%20Difference,set%20up%20the%20computer%E2%80%9D).
```suggestion
    - name: set up go env
```